src/ExadelMentorship.Tests/Features/WeatherTest.cs

```diff
@@ -0,0 +1,44 @@
+﻿using ExadelMentorship.BusinessLogic.Exceptions;
+using ExadelMentorship.BusinessLogic.Features.WeatherFeature;
+using ExadelMentorship.BusinessLogic.Models;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace ExadelMentorship.Tests.Features
+{
+    public class WeatherTest
+    {
+        [Fact]
+        public async Task GetTemperatureByCityName_WhenModelIsValid_ReturnsCity()
+        {
+            //Arrange
+            City city = new City()
+            {
+                Name = "Tbilisi"
+            };
+            Weather weather = new Weather();
+
+            //Act
+            var result = await weather.GetTemperatureByCityName(city);
```

Comment:
1. This is not a unit test, since there is a real http request. You have to be able to [mock HttpClient](https://stackoverflow.com/questions/36425008/mocking-httpclient-in-unit-tests). Also, you will most likely make HttpClient explicit dependency of Weather class. Consider constructor injection.

2. The function takes City and returns City object. Obviously this is misleading. Make the method more self-explanatory. For instance, takes city name, returns temperature.

Reply:
Done